utils/perl5lib/Config/ConfigCompsetGrid.pm

```diff
@@ -472,19 +472,19 @@ sub setCompsetGeneralVars
       $compset_longname = rmCompsetOptComps($compset_longname);
       @nodes = $xml_compset->findnodes(".//compset[lname=\"$compset_longname\"]");
     }
-    if (! @nodes) {die " ERROR: $compset_longname not supported \n";}
-
-    my $compset_grid = $nodes[0]->getAttribute('grid');
-    if ($compset_grid) {
-	$compset_grid = _clean($compset_grid);
-	if ($grid_longname !~ /$compset_grid/) {
-	    # TODO - this is CESM specific and should be removed
-	    if ($grid_longname !~ /CLM_USRDAT/) {
-		die "ERROR: $compset_longname \n is not supported for \n $grid_longname \n";
+#    if (! @nodes) {die " ERROR: $compset_longname not supported \n";}
```

Comment:
So this completely removes the check for whether compset_longname is supported. I don't understand this code well enough to know what the purpose of this check was. On the surface, it seems like we shouldn't be doing that check if we want to support -user_compset, but somehow it worked to have this check in place in recent versions of cime. So it's hard for me to say whether it's acceptable to simply remove it.

I'd like to get sign-off from @mvertens here, since she's the one who added this code originally.

Also, if it is acceptable to just remove it, then it would be good to remove this line rather than leaving it commented-out.